### DISCUSSION TITLE
docker login to public repo: public.ecr.aws or public.ecr.aws/<registry-alias>

### DISCUSSION THREAD

**Discussion by user706** [aws/aws-cli#7615] [Q&A]
Hi,

if I go to https://eu-central-1.console.aws.amazon.com/ecr/repositories?region=eu-central-1 and click **Public** and select a image-repo and click "View push commands", then I can read

> Retrieve an authentication token and authenticate your Docker client to your registry.
> Use the AWS CLI:
> 
> ```
> aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws/z3o4p8r4
> ```

Look at the last string above: it is `public.ecr.aws/z3o4p8r4`, i.e. `public.ecr.aws/<registry-alias>`
I do not think that this is correct. (At least not for docker-credential-pass (as available in debian [here](https://packages.debian.org/bullseye/golang-docker-credential-helpers))).

Because 

### Bad flow with  `public.ecr.aws/z3o4p8r4`

`~/.docker/config.json`
```json
{
  "auths": {},
  "credsStore": "desktop",
  "credHelpers": {
    "public.ecr.aws/z3o4p8r4": "pass"
  }
}
```

Because if I do
```bash
aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws/z3o4p8r4
#                                                                                                                  --------
#                                                                                                                      |
#                                                                                         see this    ----------------/
```
followed by
```bash
docker logout public.ecr.aws/z3o4p8r4
```

then I can still see

`~/.docker/config.json`
```json
{
  "auths": {
    "public.ecr.aws": {}           // this!!!  it's not really logged out!
  },
  "credsStore": "desktop",
  "credHelpers": {
    "public.ecr.aws/z3o4p8r4": "pass"
  }
}
```

Better is the following:

### Good flow with  `public.ecr.aws` (not `public.ecr.aws/z3o4p8r4`)

`~/.docker/config.json`
```json
{
  "auths": {},
  "credsStore": "desktop",
  "credHelpers": {
    "public.ecr.aws": "pass"
  }
}
```

Now if I do
```bash
aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws
#                                                                                                                  --------
#                                                                                                                      |
#                                                                                  no repository alias ---------------/
```
followed by
```bash
docker logout public.ecr.aws
```

then I can now see

`~/.docker/config.json`
```json
{
  "auths": {},                    // great! correctly logged out!
  "credsStore": "desktop",
  "credHelpers": {
    "public.ecr.aws": "pass"
  }
}
```

Thus the registry-alias, should not be part of the login. 

Is this correct? Thanks.

**Comment by user706** [comment]
The green code snippet here:
https://docs.aws.amazon.com/AmazonECR/latest/public/getting-started-cli.html#cli-authenticate-registry

shows

```bash
aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws
```

which also suggest that using something like `public.ecr.aws/<your-registry-alias>` is not needed.

**Comment by uldyssian-sh** [comment]
When using Amazon ECR Public, you must authenticate Docker only against the registry host public.ecr.aws, not against the registry alias (public.ecr.aws/<alias>).

The registry alias is part of the image path, not the login target.

Correct login:

aws ecr-public get-login-password --region us-east-1 | \
docker login --username AWS --password-stdin public.ecr.aws

Correct image usage:

docker pull public.ecr.aws/<registry-alias>/<repository>:<tag>

Why this matters:
	•	Docker stores credentials per registry host
	•	Logging in to public.ecr.aws/<alias> creates incorrect credential entries
	•	This can break docker logout and credential helpers

**Reply by uldyssian-sh** [reply]
Thanks! If this helps, please consider marking it as “Helpful” or “Accepted” 😊
